src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -1,5055 +0,0 @@
-// Copyright (c) Microsoft Corporation.

```

Comment:
I've written brand new tests in `VtIoTests.cpp`. They assert that calling a console API a certain way produces a given VT output.

Reply:
Sure, but like, this file has like 50 examples of weird wacky edge cases of combining a bunch of calls, and making sure that the host and terminal are in agreement. I'd hate to lose all those tests - some for particularly gnarly bugs. Especially the resizing ones.

Reply:
Unfortunately, almost all tests assert the current VT-to-VT translation. Since we pretty much don't modify VT output anymore, most tests cannot possibly apply anymore. 😣

Here are the ones I found that we can port:
* TestResizeHeight
  ❔ I'm not entirely sure how to translate the test though, as it asserts that VtEngine produces a given output on resize. This PR however never emits any output on resize by design, so what output would we test?
* ResizeRepaintVimExeBuffer
  ❔ Similar to above, we don't emit any output on resize anymore, so I'm not sure how to port this.
* ClsAndClearHostClearsScrollbackTest
  ✖️ Partially covered by `VtIoTests::ScrollConsoleScreenBufferW`.
  ✅ I forgot to test this in `VtIoTests::FillConsoleOutputCharacterW`.
* ClearBufferSignal
  ✅ This one should now assert that no output is produced.

<details>
<summary>These are the ones we cannot port over</summary>

* ConptyOutputTestCanary
  ❌ Unit test for VtEngine.
* SimpleWriteOutputTest
  ❌ Unit test for VtEngine.
* WriteTwoLinesUsesNewline
  ❌ Tests if a text-CUP-text sequence is translated to text-CRLF-text. With VT passthrough we won't do that anymore.
* WriteAFewSimpleLines
  ✖️ Covered by `VtIoTests::WriteConsoleW` (which should be extended to cover more inputs to be fair)
* PassthroughClearScrollback
  ❌ Tests if CSI 3 J is passed through.
* PassthroughClearAll
  ❌ Tests if CSI 2 J is passed through.
* PassthroughHardReset
  ❌ Tests if ESC c is passed through.
* PassthroughCursorShapeImmediately
  ❌ Tests if DECSCUSR is passed through.
* PassthroughDECCTR
  ❌ Tests if DECCTR is passed through.
* TestWrappingALongString
  ✖️ Covered by `ScreenBufferTests::MultilineWrap` among others.
* TestAdvancedWrapping
  ❌ Unit test for VtEngine.
* TestExactWrappingWithoutSpaces
  ✖️ Covered by `VtIoTests::WriteConsoleW`.
* TestExactWrappingWithSpaces
  ✖️ Covered by `VtIoTests::WriteConsoleW`.
* MoveCursorAtEOL
  ❌ Tests if a CUP + CSI K is translated to space + CUP. With VT passthrough we won't do that anymore.
* OutputWrappedLinesAtTopOfBuffer
  ❌ Unit test for VtEngine & covered by `ScreenBufferTests::MultilineWrap` among others.
* OutputWrappedLinesAtBottomOfBuffer
  ❌ Unit test for VtEngine.
* ScrollWithChangesInMiddle
  ❌ Unit test for VtEngine (a combination of the previous 2 tests).
* DontWrapMoveCursorInSingleFrame
  ❌ Tests if a mix of VT is passed through.
* ClearHostTrickeryTest
  ❌ Unit test for VtEngine (alternate screen buffer, etc.).
* OverstrikeAtBottomOfBuffer
  ❌ Tests if a mix of VT is passed through.
* MarginsWithStatusLine
  ❌ Tests if a mix of VT is passed through.
* OutputWrappedLineWithSpace
  ❌ Unit test for VtEngine (ensures that a space can wrap a line).
* OutputWrappedLineWithSpaceAtBottomOfBuffer
  ❌ Unit test for VtEngine (ensures that a space can wrap a line).
* BreakLinesOnCursorMovement
  ❌ Tests if a mix of VT is passed through.
* ScrollWithMargins
  ❌ Tests if a mix of VT is passed through.
* TestCursorInDeferredEOLPositionOnNewLineWithSpaces
  ❌ Asserts behavior of VT sequences (cannot break with VT passthrough).
* TestResizeWithCookedRead
  ✖️ Asserts that cooked read doesn't crash.
* NewLinesAtBottomWithBackground
  ❌ Tests if a mix of VT is passed through.
* WrapNewLineAtBottom
  ❌ Tests if a mix of VT is passed through.
* WrapNewLineAtBottomLikeMSYS
  ❌ Tests if a mix of VT is passed through.
* DeleteWrappedWord
  ❌ Tests if a mix of VT is passed through.
* HyperlinkIdConsistency
  ❌ Tests if a mix of VT is passed through.
* ResizeInitializeBufferWithDefaultAttrs
  ✖️ I think this is covered by `ScreenBufferTests::TestReflowEndOfLineColor`.
* SimpleAltBufferTest
  ❌ Tests if a mix of VT is passed through.
* AltBufferToAltBufferTest
  ❌ Tests if a mix of VT is passed through.
* TestPowerLineFirstFrame
  ❌ Tests if a mix of VT is passed through.
* AltBufferResizeCrash
  ❌ Tests if a mix of VT is passed through.
* TestNoExtendedAttrsOptimization
  ❌ Unit test for VtEngine.
* TestNoBackgroundAttrsOptimization
  ❌ Unit test for VtEngine.
* SimplePromptRegions
  ❌ Tests if a mix of VT is passed through.
* MultilinePromptRegions
  ❌ Tests if a mix of VT is passed through.
* ManyMultilinePromptsWithTrailingSpaces
  ❌ Tests if a mix of VT is passed through.
* ReflowPromptRegions
  ❌ Tests if a mix of VT is passed through.
* ClearMarksTest
  ❌ Tests if a mix of VT is passed through.

</details>